File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0, right = nums.length - 1;

        int foundIndex = -1;

        while(right >= left){
            int mid = (left + right)/2;
            if(nums[mid] <= nums[nums.length - 1]){
                foundIndex = mid;
                right = mid - 1;
            }else{
                left = mid + 1;
            }
        }

        return nums[foundIndex];
    }
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        element_list = []

        if root is None:
            return -1

        def kthSmallestDfs(root : Optional[TreeNode]) -> None:
            if root is None:
                return 

            if len(element_list) >= k:
                return 
            
            # Iterate over left child elements
            kthSmallestDfs(root.left)

            element_list.append(root)

            if len(element_list) >= k:
                return 

            # Iterate over right child elements
            kthSmallestDfs(root.right)
        
        kthSmallestDfs(root)

        return element_list[k-1].val


        